R/method-reroot.R

```diff
@@ -22,3 +22,22 @@ setMethod("reroot", signature(object="phylo"),
               return(tree)
           })
 
+
+##' @rdname reroot-methods
+##' @exportMethod reroot
+setMethod("reroot", signature(object="treedata"),
+        function(object, node, ...) {
+        	# reroot tree
+            tree <- object@phylo
+            tree <- reroot(tree, node, ...)
+            object@phylo <- tree
+            
+            # update node numbers in data
+            n.tips <- length(tree$tip.label) # Is there a better way in ggtree/treeio to get the number of tips?
+            node_map <- attr(tree, "node_map")
+            data <- object@data
+            data$node[match(node_map$from, as.integer(data$node))] <- node_map$to
```

Comment:
If object@extraInfo has information, also need to map.

Reply:
Thanks for the input Yu. I will implement all of these in some time.

Reply:
Hi Yu, Can you point me to an example of treedata with `@extraInfo`? I'm not sure what it looks like.

Reply:
similar to `@data`, and both can be empty. <https://github.com/GuangchuangYu/treeio/blob/5900d808d65f1cdd6ffdc277d361618c8c905163/R/method-drop-tip.R#L72-L84>

Reply:
Thanks. I've implemented updating `object@extraInfo` and changed the update of `object@data` to be similar.